### ISSUE TITLE
Feature Request: Set Variables inside [SETUP] with improved multiline syntax

### ISSUE THREAD

**Issue by Fitzbert-Fitz** [robotframework/robotframework#4419]
Our team prefers to keep the keywords which are run during e.g. test case setup above the testcase. We do that to keep the setup close to the test case.
```
*** Test Cases ***
Test 1
    [SETUP]    run keywords     some keyword
    ...  AND  some more keywords
    ...  AND  some more keywords
```

We would love to see the [SETUP] being more powerful.
```
*** Test Cases ***
Test 1
    [SETUP]    ${might}=     prepare something and return result
    [SETUP]    ${be}=     prepare something and return result
    [SETUP]    ${usefull}=     prepare something and return result
    test execution
```
I am not proposing this exact syntax. But you get the point

Advantages:
- variables would not show up magically in testcases coming from keywords which are defined somewhere else
- avoids the use of "Set Variable" which in my opinion is wasted space ;-)
- readability of testcases would be improved. You do not need to jump between *** Test Cases *** and *** Keywords *** to understand what is happening inside the setup for preparing the test case
- multiline setups would be easier to read avoiding the clunky "run keywords ... AND" syntax
- most new robot users I am onboarding are confused by the fact, that they can't do "[SETUP]   ${variable}=    something". I have the feeling it would be more intuitive

Maybe these are two new features:
1. enable easier to read multiline setup using an improved syntax
2. enable setting variables inside setups

Let me know if I should split it into two. 

Cheers Martin

**Comment by pekkaklarck** [comment]
Being able to run multiple keywords as a setup without defining a new one-off keyword or using `Run Keywords` would be nice, but we'd need to figure out some good and generic syntax for it. Just using `[Setup]` multiple times doesn't look too good to me. Some kind of a block syntax that could also be used in other cases would be more convenient.

The reason you cannot have

**Comment by Fitzbert-Fitz** [comment]
I totally agree that multiple [Setup]/[Teardown] lines are not a beautiful solution and I did not want to propose this syntax. In general a block syntax for Setup and Teardowns would be very helpful. And if variable assignment inside [Setup] comes as a side effect, I would love to see that.

Maybe it would also be an Idea to add and [ACT] statement which explicitly starts the test actions. So everything between [Setup] and [ACT] is considered as "setup" steps.

```
*** Test Cases ***
Test 1
    [SETUP]    ${might}=     prepare something and return result
    ${be}=     prepare something else and return result
    ${useful}=     prepare a pie and return result
    test execution
    [ACT]     Testing something
```
But I am afraid that this would lead to the ... AND syntax again.

**Comment by pekkaklarck** [comment]
I don't like `[ACT]` a marker to end the setup too much. It's not that easy to understand and it would also require big changes to the parser without being that generic. If we were to use a marker like that, we would need explicit start and end markers similarly as we have
```
IF    $cond
    ${x} =    kw 1
    kw 2
END
```

Something like
```
[Setup]    BLOCK
${x} =    kw 1
kw 2
END
kw in test
```
could possibly work. There possibly could be an inline variant similarly as we have inline IF. It could look something like this
```
[Setup]    BLOCK    ${x} =    kw 1    AND    kw 2
```

Instead of explicit start/end markers we could also consider something like
```
[Setup]
| ${x} =    kw 1
| kw 2
kw in test
```
but that would be more work to implement.

**Comment by pekkaklarck** [comment]
Implementing the BLOCK syntax would be a somewhat big task in general, but it would be even more work to support it also with setups and teardowns. They currently only work with keywords, but we have already earlier discussed that supporting inline IF with them would be handy. For example, this doesn't currently work:

```
[Setup]    IF    $cond    kw 1    ELSE    kw 2
```

**Comment by Fitzbert-Fitz** [comment]
A cleaner block Syntax would the cherry on top. But the following would actually do the job. Maybe that is easier to implement?
```
TestCase
    [Setup]  Run keywords     ${var} =    some keyword return value
    …   AND    ${var2}=      some other return value
    kw     ${var}
    kw2  ${var2}
```
and for a one line setup

```
TestCase
    [Setup]   ${var} =    some keyword return value
    kw     ${var}
```

**Comment by pekkaklarck** [comment]
The problem with assignment as part of `Run Keywords` as well as with setup is that both
```
Run Keywords    ${kw}    arg    AND    ${kw}    arg
```
and
```
[Setup]    ${kw}    arg
```
are currently supported so that `${kw}` is the name of the keyword to execute. We could require making `=` mandatory with assignment in this case so that
```
Run Keywords    ${var} =    kw    AND    ${var} =    kw
```
and
```
[Setup]    ${var} =    kw
```
would be parsed differently than the above examples. I don't like that too much because `=` isn't mandatory with assignments otherwise. It would also break this contrived but anyway currently working example:

```robotframework
*** Variables ***
${x}        kw

*** Test Cases ***
Example
    [Setup]    ${x} =    arg
    No Operation

*** Keywords ***
kw =
    [Arguments]    ${arg}
    Log    ${arg}
```
